auth/authpayload/http.go

```diff
@@ -0,0 +1,368 @@
+
+	includedHeaders, err := extractHeadersToInclude(res.Header, IsHeaderToIncludeInResponse)
+	if err != nil {
+		return nil, fmt.Errorf("failed to build payload from response headers: %w", err)
+	}
+
+	writer.WriteVarInt(uint64(len(includedHeaders)))
+
+	for _, header := range includedHeaders {
+		writer.WriteString(header.Name)
+		writer.WriteString(header.Value)
+	}
+
+	writer.WriteIntBytesOptional(res.Body)
+
+	return writer.Buf, nil
+}
+
+func ToSimplifiedHttpResponse(payload []byte, opts ...func(*HttpResponseDeserializationOptions)) (requestID []byte, res SimplifiedHttpResponse, err error) {
+	options := &HttpResponseDeserializationOptions{}
+	for _, opt := range opts {
+		opt(options)
+	}
+
+	responseReader := util.NewReader(payload)
+
+	requestID, err = responseReader.ReadBytes(32)
+	if err != nil {
+		return nil, res, fmt.Errorf("failed to read response to create http response: %w", err)
+	}
+
+	statusCode, err := responseReader.ReadVarInt32()
+	if err != nil {
+		return nil, res, fmt.Errorf("failed to read status code to create http response: %w", err)
+	}
+
+	nHeaders, err := responseReader.ReadVarInt32()
+	if err != nil {
+		return nil, res, fmt.Errorf("failed to read header count to create http response: %w", err)
+	}
+	responseHeaders := make(http.Header, nHeaders)
+	for i := uint32(0); i < nHeaders; i++ {
+		var headerKey, headerValue string
+
+		headerKey, err = responseReader.ReadString()
+		if err != nil {
+			return nil, res, fmt.Errorf("failed to read header key to create http response: %w", err)
+		}
+
+		headerValue, err = responseReader.ReadString()
+		if err != nil {
+			return nil, res, fmt.Errorf("failed to read header value to create http response: %w", err)
+		}
+
+		responseHeaders.Add(headerKey, headerValue)
+	}
+
+	if options.senderPublicKey != nil {
+		responseHeaders.Set(brc104.HeaderIdentityKey, options.senderPublicKey.ToDERHex())
+	}
+
+	responseBody, err := responseReader.ReadOptionalBytes()
+	if err != nil {
+		return nil, res, fmt.Errorf("failed to read body: %w", err)
+	}
+
+	res.StatusCode = int(statusCode)
+	res.Header = responseHeaders
+	res.Body = responseBody
+
+	return requestID, res, nil
+}
+
+func IsHeaderToIncludeInRequest(headerName string) bool {
+	headerName = strings.ToLower(headerName)
+	return isBSVHeaderToInclude(headerName) || slices.Contains(brc104.NonXBSVIncludedRequestHeaders, headerName)
+}
+
+func IsHeaderToIncludeInResponse(headerName string) bool {
+	headerName = strings.ToLower(headerName)
+	return isBSVHeaderToInclude(headerName) || slices.Contains(brc104.NonXBSVIncludedResponseHeaders, headerName)
+}
+
+func isBSVHeaderToInclude(headerName string) bool {
+	return !strings.HasPrefix(headerName, brc104.AuthHeaderPrefix) && strings.HasPrefix(headerName, brc104.XBSVHeaderPrefix)
+}
+
+func extractHeadersToInclude(headers http.Header, headersFilter func(headerName string) bool) ([]includedHeader, error) {
+	includedHeaders := make([]includedHeader, 0)
```

Comment:
You don't have to `make` the slice.
```suggestion
	var includedHeaders []includedHeader
```

But my another thought. Shouldn't you use standard map instead of slice custom `includedHeader` structs?
The map would also check for duplicates 🤔

Reply:
1. Headers are already map, so there is no possiblity to get duplicates from it
2. Actually this approach opens an easy way for handling multi values in header (but I don't know how to serialize them, that's why there is an error for `len(values) > 1` (few lines below)
3. there was also some reason why I moved from map to this approach, but I totally don't remember - maybe this was something with sorting of the headers later - that I would need to create a map first, and then a slice to have sorted headers.